packages/components/styled-system/src/define-styles.ts

```diff
@@ -12,11 +12,13 @@ export type StyleFunctionProps = {
   [key: string]: any
 }
 
-export type SystemStyleFunction = (
-  props: StyleFunctionProps,
+export type SystemStyleFunction<T extends StyleFunctionProps> = (
```

Comment:
```suggestion
export type SystemStyleFunction<T extends StyleFunctionProps = StyleFunctionProps> = (
```

Reply:
So I and @ajitjha393 discussed this change. For me, on webstorm, I didn't get any TS errors but vscode rightly pointed it out. Just doing this will break existing API.

Reply:
<img width="814" alt="image" src="https://user-images.githubusercontent.com/313363/195781961-d8508589-3e67-4983-a7fe-608ab4177ec5.png">

Reply:
If I understand it correctly, there's no good way to pass the type information received from the dynamically named variant function to  `SystemStyleInterpolation`. Is that correct @ajitjha393 ?

Reply:
Yeah as of now, I can't think of an easy way we can capture the type Paramter and make `SystemStyleInterpolation`  generic in `[variant: string] ?: SystemStyleInterpolation`

Reply:
https://github.com/chakra-ui/chakra-ui/issues/6796#issuecomment-1281831615